### PR TITLE
[VDO-5637] Updated vdorecover to work on LVMVDO volumes

### PR DESCRIPTION
**Pull request by rhawalsh**
RHEL Jira: VDO-20273

### CODE REVIEW

File: src/tools/vdorecover/vdorecover [hunk 1]
```diff
@@ -242,7 +242,8 @@ _repointUpperDevicesOrMountVDO(){
 		dmsetup resume $VDO_DEPENDENT_DEV
 		echo "You may want to remount, and run fstrim, on any filesystem"
 	        echo "mounted on ${VDO_DEPENDENT_DEV}."
-		_fstrimAndPrompt "the device or filesystem atop ${VDO_DEPENDENT_DEV}"
+                DEPENDENT_MOUNT=$(awk "/${VDO_DEPENDENT_DEV}/ {print \$2}" /proc/self/mounts)
```
Comment (slegendr) [inline src/tools/vdorecover/vdorecover:245]: Will the trailing '/' always be there? The listings I see in /proc/self/mounts don't have a trailing '/'.

**Comment by slegendr** [comment]
The pull request title or description should reference the VDO Jira ticket.

**Comment by rhawalsh** [comment]
Here is some output showing the version as written in this PR (aka w/o my debug code):
```
[root@vm-10-0-187-3 ~]# cp -a /usr /mnt/vdo/
[root@vm-10-0-187-3 ~]# rm -rf /mnt/vdo/usr
[root@vm-10-0-187-3 ~]# vi vdorecover
[root@vm-10-0-187-3 ~]# ./vdorecover
Usage: ./vdo_recover {path to vdo device}
[root@vm-10-0-187-3 ~]# mkdir recovery
[root@vm-10-0-187-3 ~]# LOOPBACK_DIR=$(pwd)/recovery TMPFILESZ=$((1024*1024*2)) ./vdorecover /dev/mapper/vg_name-vpool0-vpool
Recovery process started
Detecting dependent device vg_name-vdo_volume on vg_name-vpool0-vpool -- manual intervention will be required
You may want to remount, and run fstrim, on any filesystem
mounted on vg_name-vdo_volume.
Now down to just 30% used
Beginning commit of data changes
Merging, 2646016 more chunks for dm-0-merge
Merging, 2490368 more chunks for dm-0-merge
Merging, 2314240 more chunks for dm-0-merge
Merging, 2150400 more chunks for dm-0-merge
Merging, 1970176 more chunks for dm-0-merge
Merging, 1839104 more chunks for dm-0-merge
Merging, 1703936 more chunks for dm-0-merge
Merging, 1568768 more chunks for dm-0-merge
Merging, 1458176 more chunks for dm-0-merge
Merging, 1318912 more chunks for dm-0-merge
Merging, 1183744 more chunks for dm-0-merge
Merging, 1036288 more chunks for dm-0-merge
Merging, 917504 more chunks for dm-0-merge
Merging, 790528 more chunks for dm-0-merge
Merging, 663552 more chunks for dm-0-merge
Merging, 532480 more chunks for dm-0-merge
Merging, 409600 more chunks for dm-0-merge
Merging, 266240 more chunks for dm-0-merge
Merging, 147456 more chunks for dm-0-merge
Merging, 4096 more chunks for dm-0-merge
Recovery process completed, vg_name-vpool0-vpool is 30% Used
```

File: src/tools/vdorecover/vdorecover (hunk 1 above)
Comment (rhawalsh) [reply src/tools/vdorecover/vdorecover:245]: Are you talking about the `/${VDO_DEPENDENT_DEV}/`?  That's a regex.

Comment (slegendr) [reply src/tools/vdorecover/vdorecover:245]: Yeah - 0kay. When I tried it manually in a terminal for something listed in /proc/self/mounts it gave me a syntax error.

**Review by lorelei-sakai — DISMISSED** [review]
First commit message has a typo "chedk"

**Comment by slegendr** [comment]
Re-approved by me, outside of Matt's requested changes.

**Comment by rhawalsh** [comment]
> First commit message has a typo "chedk"

Commit message has been fixed.

**Review by raeburn — APPROVED** [review]

**Comment by rhawalsh** [comment]
Test that keeps failing is in UDSTest::UserJenkins, which is StumblingGet segfaulting.  This code is not touched with this MR.  I am disregarding and just going to move forward with merging.

**Pull request merged**